src/box/execute.c

```diff
@@ -553,6 +553,7 @@ sql_execute(struct sql_stmt *stmt, struct port *port, struct region *region)
 {
```

Comment:
Why should `NOW()` return the same value until a next request? I see you call it before each request even if it doesn't use the function. That would be expensive because it makes a system call. Probably even 2 in case `localtime_r()` is a system call too (I don't know).

What I proposed before was to make `datetime_now()` use `ev_loop` clock. To avoid any system calls at all. Higher precision doesn't make sense anyway - it would only be needed for high precision clock which is not the usecase of datetime. More for `fiber.clock()`. It wasn't accepted I don't remember why. But I suggest to revisit that idea.

The current solution can't guarantee `NOW()` being the same during a request because in case a request yields and another request starts, it will override that global `sql->now` field. So the first request will start returning new values. It surely can be confirmed with a test.

Reply:
Thank you for your comments! I have some answers below.

About the same value - that was specified by @unera.

As for `datetime_now()` - I believe it is better to ask @Mons and @tsafin. But anyway, I'll rewrite my solution so that `datetime_now()` doesn't get called if `NOW()` is not called.

My current solution is indeed flawed:
```
tarantool> func = {body = 'function() return box.execute([[SELECT NOW();]]).rows[1][1] end', returns = 'datetime', exports = {'SQL'}}
---
...

tarantool> box.execute([[select now(), now2(), now();]])
---
- metadata:
  - name: COLUMN_1
    type: datetime
  - name: COLUMN_2
    type: datetime
  - name: COLUMN_3
    type: datetime
  rows:
  - ['2022-04-06T09:40:33.660775+0300', '2022-04-06T09:40:33.660804+0300', '2022-04-06T09:40:33.660804+0300']
...
```

I will think about a new one.

Reply:
At the end I created a new DATETIME function - datetime_ev_now() which uses ev_now(). I use this function in SQL built-in function NOW().